test/acceptance/test_helper.rb

```diff
@@ -8,7 +8,7 @@ class AcceptanceTest < ActionDispatch::IntegrationTest
 
   self.use_transactional_tests = false
 
-  DatabaseCleaner.strategy = :truncation
+  DatabaseCleaner.strategy = :deletion
```

Comment:
Without this the database doesn't get cleaned properly. You can try locally with :truncation, it doesn't work.